### PR TITLE
[NCL-7256] Keep UI partially working even if KeyCloak is not available

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/services/keycloakService.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ class KeycloakService {
   }
 
   public isTokenExpired(): boolean {
-    return this.keycloakAuth.isTokenExpired(this.KEYCLOAK_TOKEN_MIN_EXP);
+    return this.isKeycloakAvailable && this.keycloakAuth.isTokenExpired(this.KEYCLOAK_TOKEN_MIN_EXP);
```
[the remark this answers is not in the document]
Comment (jomrazek) [reply src/services/keycloakService.ts:158]: https://github.com/project-ncl/pnc-web-ui-react/pull/84#discussion_r947865164

[the remark this answers is not in the document]
Comment (matedo1) [reply src/services/keycloakService.ts:158]: Hi @jomrazek , comment #84 is not valid anymore, it's already fixed, there is another issue

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/components/ProtectedContent/ProtectedContent.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export enum PROTECTED_TYPE {
+  Route = 'route',
+  Component = 'component',
+}
+
+interface IProtectedContentProps {
+  type: PROTECTED_TYPE;
+  role?: AUTH_ROLE;
+  title?: string;
+  hideOnError?: boolean;
```
Comment (matedo1) [inline src/components/ProtectedContent/ProtectedContent.tsx:14]: even if we use naming like ErrorPage in this context, what would you say to use just `hide` in all related places?

**Review by matedo1 — CHANGES_REQUESTED** [review]

**Review by matedo1 — APPROVED** [review]
Thanks for this feature!

**Pull request merged**